src/Store.js

```diff
@@ -0,0 +1,50 @@
+import invariant from 'invariant';
+import isPlainObject from './utils/isPlainObject';
+
+export default class Store {
+  constructor(reducer, initialState) {
+    invariant(
+      typeof reducer === 'function',
+      'Expected the reducer to be a function.'
+    );
+
+    this.state = initialState;
+    this.listeners = [];
+    this.replaceReducer(reducer);
+  }
+
+  getReducer() {
+    return this.reducer;
+  }
+
+  replaceReducer(nextReducer) {
+    this.reducer = nextReducer;
+    this.dispatch({ type: '@@INIT' });
+  }
+
+  dispatch(action) {
+    invariant(
+      isPlainObject(action),
+      'Actions must be plain objects. Use custom middleware for async actions.'
+    );
+
+    const { reducer } = this;
+    this.state = reducer(this.state, action);
+    this.listeners.forEach(listener => listener());
```

Comment:
@dbrans In fact this is how it used to be, but I felt it was redundant because the components would reach into `getState()` anyway in some cases. So there's no real reason other than making the API do less things. I don't have a strong opinion here though.